specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2021-11-20-preview/elasticsan.json

```diff
@@ -0,0 +1,2106 @@
+      "get": {
+        "tags": [
+          "ElasticSan"
+        ],
+        "operationId": "ElasticSans_ListByResourceGroup",
+        "description": "Gets a list of ElasticSan in a resource group.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Operation completed successfully; retrieved elasticSans by resource group",
+            "schema": {
+              "$ref": "#/definitions/ElasticSanList"
+            }
+          },
+          "default": {
+            "description": "Error from the RP",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ElasticSans_ListByResourceGroup_MaximumSet_Gen": {
+            "$ref": "./examples/ElasticSans_ListByResourceGroup_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}": {
+      "put": {
+        "tags": [
+          "ElasticSans"
+        ],
+        "operationId": "ElasticSans_Create",
+        "description": "Create ElasticSan.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ElasticSanNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ElasticSan"
+            },
+            "description": "Elastic San object."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Returned when the create operation is successfully completed.",
+            "schema": {
+              "$ref": "#/definitions/ElasticSan"
+            }
+          },
+          "202": {
+            "description": "Accepted -- Create request accepted; operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/ElasticSan"
+            }
```

Comment:
Returning 202 is okay. Note that 201 + provisioningState is a preferred pattern.

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

Reply:
Thanks @j5lim . We will be addressing this suggestion in our upcoming PR. Created a tracking item in our backlog: https://msazure.visualstudio.com/One/_workitems/edit/14150911